deeplake/core/vectorstore/deep_memory.py

```diff
@@ -0,0 +1,348 @@
+        self,
+        dataset: Dataset,
+        embedding_function: Optional[Any] = None,
+        token: Optional[str] = None,
+    ):
+        """Based Deep Memory class to train and evaluate models on DeepMemory managed service.
+
+        Args:
+            dataset (Dataset): deeplake dataset object.
+            embedding_function (Optional[Any], optional): Embedding funtion class used to convert queries/documents to embeddings. Defaults to None.
+            token (Optional[str], optional): API token for the DeepMemory managed service. Defaults to None.
+
+        Raises:
+            ImportError: if indra is not installed
+        """
+        self.dataset = dataset
+        self.token = token
+        self.embedding_function = embedding_function
+        self.client = DeepMemoryBackendClient(token=token)
+        self.queries_dataset = deeplake.dataset(
+            self.dataset.path + "_queries",
+            token=token,
+            read_only=False,
+        )
+        if len(self.queries_dataset) == 0:
+            self.queries_dataset.commit(allow_empty=True)
+
+    def train(
+        self,
+        queries: List[str],
+        relevance: List[List[Tuple[str, int]]],
+        embedding_function: Optional[Callable[[str], np.ndarray]] = None,
+        token: Optional[str] = None,
+    ):
+        """Train a model on DeepMemory managed service.
+
+        Args:
+            queries (List[str]): List of queries to train the model on.
+            relevance (List[List[Tuple[str, int]]]): List of relevant documents for each query.
+            embedding_function (Optional[Callable[[str], np.ndarray]], optional): Embedding funtion used to convert queries to embeddings. Defaults to None.
+            token (str, optional): API token for the DeepMemory managed service. Defaults to None.
+
+        Returns:
+            str: job_id of the training job.
+
+        Raises:
+            ValueError: if embedding_function is not specified either during initialization or during training.
+        """
+        # TODO: Support for passing query_embeddings directly without embedding function
+        corpus_path = self.dataset.path
+        queries_path = corpus_path + "_queries"
+
+        if embedding_function is None and self.embedding_function is None:
+            raise ValueError(
+                "Embedding function should be specifed either during initialization or during training."
+            )
+
+        if embedding_function is None and self.embedding_function is not None:
+            embedding_function = self.embedding_function.embed_documents
+
+        queries_vs = VectorStore(
+            path=queries_path,
+            overwrite=True,
+            runtime={"tensor_db": True},
+            embedding_function=embedding_function,
+            token=token,
+        )
+
+        queries_vs.add(
+            text=[query for query in queries],
+            metadata=[
+                {"relevance": [(doc_id, 1) for doc_id in relevance]}
```

Comment:
relevance is already a a tuple list (id, 1) so if you iterate that would be ((id, 1), 1) which is incorrect. maybe wrong but you need to change this to be a
`{"relevance": relevance}`

Reply:
Didn't we decide to shift from` List[List[Tuple[str, int]]]` to `List[List[str]]`?